CMakeLists.txt

```diff
@@ -427,6 +427,19 @@ if (LIBCRYPTO_SUPPORTS_EVP_MD_CTX_SET_PKEY_CTX)
     target_compile_options(${PROJECT_NAME} PUBLIC -DS2N_LIBCRYPTO_SUPPORTS_EVP_MD_CTX_SET_PKEY_CTX)
 endif()
 
+# Determine if Kyber512 implementation from AWS-LC is available
+try_compile(
+        LIBCRYPTO_SUPPORTS_EVP_KYBER_512
+        ${CMAKE_BINARY_DIR}
+        SOURCES "${CMAKE_CURRENT_LIST_DIR}/tests/features/evp_kyber_512.c"
+        LINK_LIBRARIES ${LINK_LIB} ${OS_LIBS}
+        COMPILE_DEFINITIONS "-Werror"
+)
+
+if(S2N_AWSLC_KYBER_UNSTABLE AND LIBCRYPTO_SUPPORTS_EVP_KYBER_512)
+    target_compile_options(${PROJECT_NAME} PUBLIC -DAWS_LC_SUPPORTS_KYBER_512)
```

Comment:
While it is currently true, AWS-LC might not be the only libcrypto that supports this EVP mode in the long term. I would prefer to keep this generic to something like `S2N_LIBCRYPTO_SUPPORTS_KYBER512`